mlir/test/Dialect/Mesh/forward-sharding-propagation.mlir

```diff
@@ -0,0 +1,49 @@
+// RUN: mlir-opt --pass-pipeline="builtin.module(func.func(sharding-propagation{traversal=forward}))" %s | FileCheck %s
```

Comment:
I think this means only two out of four traversals are checked, right? The default `backward-forward` (presumably in existing tests) and just `forward` in this test?

Is it worth adding tests for the other traversals?

Reply:
Basically the backward traversal is exercised in the default case, just followed by a forward pass.
I didn't have a case ready that require the other traversals. I could make something up. Not sure how useful that would be.